### PR TITLE
[ISSUE-2730] Add the number of unseq merge times in TsFile name.

### PR DESCRIPTION
**Pull request by WilliamSong11**
## Description
Hi all :
I have submitted a pr for [ISSUE-2730](https://github.com/apache/iotdb/issues/2730), please take a look.
This issue about add the number of unseq merge times in TsFile name.

The modules involved are as follows:
   1、server/engine/merge
   2、server/engine/storagegroup
### Content3 ...

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

**Comment by WilliamSong11** [comment]
@HTHou  Hi haonan, I have submitted a pr for  "Add the number of unseq merge times in TsFile name" please take a look.

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/level/LevelCompactionTsFileManagement.java [hunk 1]
```diff
@@ -600,14 +601,13 @@ private void merge(
...
-            CompactionLogger compactionLogger =
-                new CompactionLogger(storageGroupDir, storageGroupName);
+            compactionLogger = new CompactionLogger(storageGroupDir, storageGroupName);
             // log source file list and target file for recover
             for (TsFileResource mergeResource : mergeResources.get(i)) {
               compactionLogger.logFile(SOURCE_NAME, mergeResource.getTsFile());
             }
             File newLevelFile =
-                createNewTsFileName(mergeResources.get(i).get(0).getTsFile(), i + 1);
+                TsFileResource.modifyTsFileNameMergeCnt(mergeResources.get(i).get(0).getTsFile());
```
[the remark this answers is not in the document]
Comment (WilliamSong11) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/level/LevelCompactionTsFileManagement.java:610]: image
Thank you for reply，We really need level info to decide whether  to compact the tsfile，<TsFileResource.modifyTsFileNameMergeCnt> This function increments the number of sequential merges by 1，level info taken from mergeResource。

[the remark this answers is not in the document]
Comment (mychaow) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/level/LevelCompactionTsFileManagement.java:610]: got it.

**Review by mychaow — APPROVED** [review]
LGTM

**Comment by WilliamSong11** [comment]
@HTHou Hi  haonan  Are there any adjustments we need to make

**Review by HTHou — APPROVED** [review]

**Pull request merged**